implementation/demos/compare_cuas_snes.py

```diff
@@ -0,0 +1,164 @@
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--theta", default=1, type=np.float64, dest="theta",
+                        help="Theta parameter for Nitsche, 1 symmetric, -1 skew symmetric, 0 Penalty-like")
+    parser.add_argument("--gamma", default=1000, type=np.float64, dest="gamma",
+                        help="Coercivity/Stabilization parameter for Nitsche condition")
+    _solve = parser.add_mutually_exclusive_group(required=False)
+    _solve.add_argument('--linear', dest='linear_solver', action='store_true',
+                        help="Use linear solver", default=False)
+    _3D = parser.add_mutually_exclusive_group(required=False)
+    _3D.add_argument('--3D', dest='threed', action='store_true',
+                     help="Use 3D mesh", default=False)
+    _strain = parser.add_mutually_exclusive_group(required=False)
+    _strain.add_argument('--strain', dest='plane_strain', action='store_true',
+                         help="Use plane strain formulation", default=False)
+    _dirichlet = parser.add_mutually_exclusive_group(required=False)
+    _dirichlet.add_argument('--dirichlet', dest='dirichlet', action='store_true',
+                            help="Use strong Dirichlet formulation", default=False)
+    _E = parser.add_argument("--E", default=1e3, type=np.float64, dest="E",
+                             help="Youngs modulus of material")
+    _nu = parser.add_argument(
+        "--nu", default=0.1, type=np.float64, dest="nu", help="Poisson's ratio")
+    _disp = parser.add_argument("--disp", default=0.08, type=np.float64, dest="disp",
+                                help="Displacement BC in negative y direction")
+    _ref = parser.add_argument("--refinements", default=1, type=np.int32,
+                               dest="refs", help="Number of mesh refinements")
+    _gap = parser.add_argument(
+        "--gap", default=0.02, type=np.float64, dest="gap", help="Gap between plane and y=0")
+
+    # Parse input arguments or set to defualt values
+    args = parser.parse_args()
+
+    # Current formulation uses unilateral contact
+    nitsche_parameters = {"gamma": args.gamma, "theta": args.theta}
+    nitsche_bc = not args.dirichlet
+    physical_parameters = {"E": args.E, "nu": args.nu, "strain": args.plane_strain}
+    vertical_displacement = -args.disp
+    num_refs = args.refs + 1
+    gap = args.gap
+    top_value = 1
+    threed = args.threed
+    bottom_value = 2
+
+    # Load mesh and create identifier functions for the top (Displacement condition)
+    # and the bottom (contact condition)
+    if threed:
+        fname = "sphere"
+        create_sphere_mesh(filename=f"{fname}.msh")
+        convert_mesh(fname, "tetra")
+        with dolfinx.io.XDMFFile(MPI.COMM_WORLD, f"{fname}.xdmf", "r") as xdmf:
+            mesh = xdmf.read_mesh(name="Grid")
+        # mesh = dolfinx.UnitCubeMesh(MPI.COMM_WORLD, 10, 10, 20)
+
+        # def top(x):
```

Comment:
Could we remove code that is commented. If we want to keep these configurations, it is better to add an argument to the argparser.